SchedulerBot/SchedulerBot.Infrastructure.BotConnector/MessageProcessor.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using SchedulerBot.Business.Entities;
+using SchedulerBot.Database.Entities;
+using SchedulerBot.Infrastructure.Interfaces.BotConnector;
+
+namespace SchedulerBot.Infrastructure.BotConnector
+{
+	public class MessageProcessor : IMessageProcessor
+	{
+		private readonly AppCredentials appCredentials;
+		private readonly ILogger<MessageProcessor> logger;
+
+		public MessageProcessor(
+			AppCredentials appCredentials,
+			ILogger<MessageProcessor> logger)
+		{
+			this.appCredentials = appCredentials;
+			this.logger = logger;
+		}
+
+		public async Task SendMessageAsync(ScheduledMessage scheduledMessage, CancellationToken cancellationToken)
+		{
+			MicrosoftAppCredentials credentials = BuildMicrosoftAppCredentials();
+
+			Uri serviceUri = new Uri(scheduledMessage.Details.ServiceUrl);
+			Activity activity = CreateBotMessageActivity(scheduledMessage);
+
+			using (ConnectorClient connector = new ConnectorClient(serviceUri, credentials))
+			{
+				logger.LogInformation($"Sending message to conversation RecipientId: ${activity.Recipient.Id}.");
+				await connector.Conversations.SendToConversationAsync(activity, cancellationToken);
+			}
+		}
+
+		public Task<ResourceResponse> ReplyAsync(Activity activity, string replyText)
```

Comment:
It might make sense to take a CancellationToken for consistency with `SendMessageAsync`.